### ISSUE TITLE
E208 raised on {file: {path: foo, state: file}}

### ISSUE THREAD

**Issue by moreati** [ansible/ansible-lint#1005]
##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
:~/venvs$ ./ansible-2.9.12/bin/ansible --version
ansible 2.9.12
  config file = /home/alex/.ansible.cfg
  configured module search path = ['/home/alex/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/alex/venvs/ansible-2.9.12/lib/python3.8/site-packages/ansible
  executable location = ./ansible-2.9.12/bin/ansible
  python version = 3.8.2 (default, Jul 16 2020, 14:00:26) [GCC 9.3.0]

:~/venvs$ ./ansible-2.9.12/bin/ansible-lint --version
ansible-lint 4.3.3.dev1+gcc23bae

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
Ubuntu 20.04 x86_64

##### STEPS TO REPRODUCE

```console (paste below)
~/venvs$ cat playbook.yml 
- name: Test E208
  hosts: localhost
  tasks:
    - name: File module without mode
      file:
        path: canary
        state: file

:~/venvs$ ./ansible-2.9.12/bin/ansible-lint -p playbook.yml 
playbook.yml:4: [E208] File permissions not mentioned
```

##### Desired Behaviour
A task that will not create a file, or change existing mode, is not flagged as needing a 'mode'.

##### Actual Behaviour
The task is flagged.

PR (probably based on #996) in progress. ETA a few hours.
